### PR TITLE
object indexing in a web worker

### PR DESCRIPTION
**Pull request by UnsungHero97**

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -40,7 +40,7 @@
   "dependencies" : {
   },
   "devDependencies" : {
-    "axios"              : "0.5.4",
+    "axios"              : "0.9.1",
```
Comment (makosblade) [inline bower.json]: More perf? :D

Comment (UnsungHero97) [inline bower.json]: maybe? but the reason for this is web worker support.

File: js/src/service/StorageClient.coffee [hunk 2]
```diff
@@ -231,18 +231,12 @@ define 'kryptnostic.storage-client', [
               @objectApi.setObjectFromBlockCiphertext(objectKeyForNewlyCreatedObject, blockCiphertext)
             .then =>
               if storageRequest.isSearchable
-                # ToDo: PLATFORM-61 - search and indexing migration to backend v2
-                return @objectIndexingService.index(
+                @objectIndexingService.enqueueIndexTask(
```
Comment (synecdoche-nh) [inline js/src/service/StorageClient.coffee]: it doesn't look like we fall back to non-worker indexing, that correct?
if so, have tested on FF, Chrome, IE11 that indexing in worker is g2g?

Comment (UnsungHero97) [inline js/src/service/StorageClient.coffee]: incorrect. we'll fall back to indexing in the main thread if web workers aren't supported, or something else causes the Promise to reject. the `.catch` in `enqueueIndexTax` will trigger the main thread indexing if that happens.

Comment (UnsungHero97) [inline js/src/service/StorageClient.coffee]: good call... I tested across browsers, and I ended up fixing a bunch of bugs.

**Pull request merged**